### PR TITLE
Change bin/rails to bin/rake where necessary in engines.md

### PR DESCRIPTION
**Pull request by nakulpathak3**
The rails guides for Engines had some issues where `bin/rails` was being called instead of `bin/rake`. I made the change to `rake` where necessary. 

Could someone also verify whether we need `rake` or `bundle exec rake` when running rake commands in the engine? For example, [here](https://github.com/rails/rails/compare/master...nakulpathak3:master#diff-3cdd8629f08c4f9af2e70edaa8e6f349R488) and [here](https://github.com/rails/rails/compare/master...nakulpathak3:master#diff-3cdd8629f08c4f9af2e70edaa8e6f349R426), I went with `rake` since the engine does not add a `bin/rake` binary by default. Should this be `bundle exec rake`?

### CODE REVIEW

**Comment by kaspth** [comment]
Thanks, but these are all correct. You should be able to call rake tasks via rails in apps and engines.

File: guides/source/engines.md [hunk 1]
```diff
@@ -423,7 +423,7 @@ Finally, the assets for this resource are generated in two files:
 `app/assets/stylesheets/blorgh/articles.css`. You'll see how to use these a little
 later.
 
-You can see what the engine has so far by running `bin/rails db:migrate` at the root
+You can see what the engine has so far by running `rake db:migrate` at the root
```
Comment (maclover7) [inline guides/source/engines.md]: bin/rake

File: guides/source/engines.md [hunk 2]
```diff
@@ -485,7 +485,7 @@ called `Blorgh::Comment`. Now run the migration to create our blorgh_comments
 table:
 
 ```bash
-$ bin/rails db:migrate
+$ rake db:migrate
```
Comment (maclover7) [inline guides/source/engines.md]: bin/rake ?

**Comment by maclover7** [comment]
Ahh, whoops. you got to this quicker than me, @kaspth !

**Comment by nakulpathak3** [comment]
I tried running these while following the guide and ran into this. I am in the directory of the gem in this case.

```
~/blorgh-engine-gem (master) [05:01:06]
nakul$ bin/rails db:migrate
Error: Command not recognized
Usage: rails COMMAND [ARGS]

The common Rails commands available for engines are:
 generate    Generate new code (short-cut alias: "g")
 destroy     Undo code generated with "generate" (short-cut alias: "d")

All commands can be run with -h for more information.

If you want to run any commands that need to be run in context
of the application, like `rails server` or `rails console`,
you should do it from application's directory (typically test/dummy).
```

**Comment by nakulpathak3** [comment]
Ran this in test-app which uses the engine - 
```
ruby-2.3.0@rails4.2 ~/test-app [05:03:42]
nakul$ bin/rails personal_blog:install:migrations
Error: Command 'personal_blog:install:migrations' not recognized
Did you mean: `$ rake personal_blog:install:migrations` ?

Usage: rails COMMAND [ARGS]

The most common rails commands are:
 generate    Generate new code (short-cut alias: "g")
 console     Start the Rails console (short-cut alias: "c")
 server      Start the Rails server (short-cut alias: "s")
 dbconsole   Start a console for the database specified in config/database.yml
             (short-cut alias: "db")
 new         Create a new Rails application. "rails new my_app" creates a
             new application called MyApp in "./my_app"

In addition to those, there are:
 destroy      Undo code generated with "generate" (short-cut alias: "d")
 plugin new   Generates skeleton for developing a Rails plugin
 runner       Run a piece of code in the application environment (short-cut alias: "r")

All commands can be run with -h (or --help) for more information.
```

Also verified that `bin/rake personal_blog:install:migrations` works

File: guides/source/engines.md (hunk 1 above)
Comment (nakulpathak3) [inline guides/source/engines.md]: see the description, I had a question relating to this - engines by default don't have `bin/rake`, they only come with the rails binary in the bin directory

**Comment by kaspth** [comment]
Which Rails version are you on?

**Comment by kaspth** [comment]
Ah, it says 4.2. These guides are for 5 and above. There you can run rake tasks via rails.

**Comment by nakulpathak3** [comment]
Oh I see! Thank you for your help :)

**Comment by nakulpathak3** [comment]
Should we add something to indicate this though? Other people might be confused by the same thing I was since by default the version of rails currently installed is 4.2.5.2. Also, I believe PR's changes would work with both versions of rails (~4 and 5) so might still be a good idea to make the change.

**Comment by senny** [comment]
@nakulpathak3 that's the reason why the guides are versioned. http://guides.rubyonrails.org/ will show you the way for the latest released version and http://edgeguides.rubyonrails.org/ what's right on `master`. If you browse the source you got to be prepared that it's not matching a released version. Since we want to promote `bin/rails` from 5.0 forward it's the right thing to document that.

**Pull request closed without merging**